### ISSUE TITLE
[BUG] (short description)

### ISSUE THREAD

**Issue by srettew** [bigtreetech/BIGTREETECH-SKR-mini-E3#658]
I just installed an E3 V3.0 on my Ender 3 Pro. This is regarding the 3 fan assignments and getting them to work correctly. I confirmed the following by checking function:

FAN0 (PC6) = Part cooling fan
FAN1 (PC7) = Hotend cooling fan
FAN2 (PB15) = Control board cooling fan

FAN0 turns on correctly by using fan speed control on TFT35
FAN1 turns on correctly when hotend temp is goes above 50 degrees and turns off when lower than 50 degrees
FAN2 is the problem. This is supposed to turn on when a driver is activated. I get nothing when I manually move any of the axis using the movement control on TFT35.

Where in the config_h file can I find PB15 and what is the correct code fix to make it work?

**Comment by radek8** [comment]
Can you send your configuration files?

**Comment by srettew** [comment]
It’s the config that is available on GitHub for Ender 3 Pro and SKR Mini E3 V3.0.

**Comment by StrawberryD24** [comment]
Having the same issue on an ender 3. The part fan won't turn on at all

**Comment by radek8** [comment]
You have to allow
#define USE_CONTROLLER_FAN
and define the pin
#define CONTROLLER_FAN_PIN PB15

**Comment by StrawberryD24** [comment]
@radek8 which file(s) need to be modified for the pin definitions? Can you share an example config?

**Comment by radek8** [comment]
file Configuration_adv.h

![image](https://user-images.githubusercontent.com/46979052/184529048-88a0d80a-26ed-4d44-9421-0c1671d5f9d0.png)

**Comment by StrawberryD24** [comment]
@radek8 
Thanks for your  for the insight but I have already done that. I have no issue with the board controller fan and hot end fan. my Issue is with the part cooling fan supposedly PC6 in the firmware.

I compared the pinout on the Box and user manual for the V3 board vs those published on their site and the firmware. Turns out on the pin diagrams shown on both the box and the user manual on their github,  the pins are listed as :
HB - PC6
E0 - PC9
Fan2 - PB15
Fan0 - PC7
Fan1 - PC0

Whereas the pins in stm32g0 defines them as
#define HEATER_0_PIN                        PC8   // "HE"
#define HEATER_BED_PIN                      PC9   // "HB"
#define FAN_PIN                             PC6   // "FAN0"
#define FAN1_PIN                            PC7   // "FAN1"
#define FAN2_PIN                            PB15  // "FAN2"
![Board_1](https://user-images.githubusercontent.com/45543079/185294877-a976a470-b906-4358-9508-249f712165cc.PNG)
![Board_zoom](https://user-images.githubusercontent.com/45543079/185294881-3c68c79d-6793-43d9-beab-8053ff767565.PNG)
![pins](https://user-images.githubusercontent.com/45543079/185296401-39d46bb9-dc3a-4065-9092-90765ab4a2de.PNG)

**Comment by StrawberryD24** [comment]
Here is the Box
I think it's a bigger issue than just firmware 
![1660798198411](https://user-images.githubusercontent.com/45543079/185296746-fc50d8e1-2474-43a0-9816-c274cfe1a12f.jpg)
![1660798198394](https://user-images.githubusercontent.com/45543079/185297149-ab29711f-b05e-4660-b3f7-b2939a4cbf74.jpg)
![1660798198421](https://user-images.githubusercontent.com/45543079/185297160-8a92e728-a3d9-4e64-a8be-061ca7d12b4d.jpg)
e

**Comment by radek8** [comment]
@StrawberryD24 
Yes, you are right, the description of the pins on the cse does not match the definition of the pins in the pin file.

@bigtreetech, can you comment on the fan pin definition issue and fix it?¨

**Comment by radek8** [comment]
if you think that there is a poorly defined set of pins in Marlin, create a ticket in the Marlin repository with a request for a correction, or propose a correction yourself.

https://github.com/MarlinFirmware/Marlin/issues

**Comment by StrawberryD24** [comment]
@radek8  thanks a mil for your help. It will be my first request for correction on GitHub hope I do it right lol

**Comment by WebSpectrum** [comment]
iS THIS XORRCT?:

**Comment by WebSpectrum** [comment]
Is this correct:

**Comment by WebSpectrum** [comment]
is this correct?:

FAN0 (PC6) = Part cooling fan
FAN1 (PC7) = Hotend cooling fan
FAN2 (PB15) = Control board cooling fan

or this:

FAN0 (PC6) = Part cooling fan
FAN1 (PC7) = Control board cooling fan
FAN2 (PB15) = Hotend cooling fan

I see a mix of information

**Comment by radek8** [comment]
@WebSpectrum 
The part cooling fan is defined in the pin file.
#define FAN_PIN PC6 // "FAN0"

You define the hot end cooling fans and control boards in the configuration file configuration_adv.h
#define E0_AUTO_FAN_PIN
#define CONTROLLER_FAN_PIN

**Comment by WebSpectrum** [comment]
Hello,

 

In the manual it shows FAN0 as PC0  ?

 



 

Regards

 

Paul Maher

Web Spectrum -  <http://www.webspectrum.co.uk> www.webspectrum.co.uk 

22+ Years Professional Web Design/Development & High-End SEO

Adobe Community Professional

Phone: 07837 903056

**Comment by radek8** [comment]
Send a link to the mentioned manual.

**Comment by WebSpectrum** [comment]
https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/hardware/BTT%20SKR%20MINI%20E3%20V3.0/Hardware/BTT%20SKR%20MINI%20E3%20V3.0%20user%20manual.pdf

 

 

Regards

 

Paul Maher

Web Spectrum -  <http://www.webspectrum.co.uk> www.webspectrum.co.uk 

22+ Years Professional Web Design/Development & High-End SEO

Adobe Community Professional

Phone: 07837 903056

**Comment by radek8** [comment]
> @radek8 děkuji mil za vaši pomoc. Bude to moje první žádost o opravu na GitHubu, doufám, že to udělám správně lol

@StrawberryD24 
Can you send a detail photo of your SKR board?
It looks like your board is not from BTT, but an imitation from another manufacturer. :-)

**Comment by radek8** [comment]
It seems that BTT has chaos in the documentation

SKR mini E3 V3.0

![image](https://user-images.githubusercontent.com/46979052/224424301-6069110d-4955-4ac2-af6d-ec3f3a5bee31.png)

SKR mini E3 V3.0
![image](https://user-images.githubusercontent.com/46979052/224424488-2323077b-d365-475e-9dae-b677c03ad682.png)

SKR mini E3 V3.0.1
![image](https://user-images.githubusercontent.com/46979052/224425090-1df0625d-182b-4845-a615-c3dbe76524ba.png)

**Comment by radek8** [comment]
This layout will probably be faulty.

![image](https://user-images.githubusercontent.com/46979052/224429886-d1108e40-737f-4cfc-be8a-0a608a5741a3.png)

**Comment by Wully616** [comment]
Just purchased today from amazon and seeing the same issue, the board documentation makes no sense.
Can someone confirm the fan pin layout?

**Comment by mianos** [comment]
I used 
```
#if ENABLED(USE_CONTROLLER_FAN)
  #define CONTROLLER_FAN_PIN PC8 // PB15
```
Based on the PCB 3.0.1 screen print above and I seem to have a 'whole system' fan as fan 3 in the menu that can be be controlled with PCM 0-100%.

The board is marked 3.0. The box is marked 3.0 but the D/C is 2023 1.6, right from the fab by the look of it.
(under the main text it says 'new' in Chinese.

**Comment by mogfink** [comment]
So I'm having a related but dissimilar issue if anyone has any insight... I'm using the io pins to control rgbw LEDs through MOSFETs, everything works except that when the board fan is on the blue LEDs light, and when I switch on the blue LEDs it tries to run the board fan for a moment until both fan and blue LEDs go off, I'm using pin pd4 for the blue LEDs and the board fan is on fan1

**Comment by Gavisann** [comment]
I have been facing the same issue with fan headers and this is what worked for me using the precompiled firmware that shipped with my Mini E3 V3 and Ender 3 Pro.

FAN0 - Part cooling fan - Confirmed by sending `M106 S255` via Octoprint's Terminal which runs the fan at 100%.
FAN1 - Hotend fan - Confirmed by heating hotend to 200c, fan kicks on at 50c
FAN2 - Controller fan - Confirmed by using Octoprint to move the X/Y motors and the fan runs for 1-2minutes.

**Comment by tim-speight** [comment]
HOW DO YOU KNOW IF YOU HAVE A 3.0 OR A 3.0.1? My boards case say's New and manufactured 2022.12.31

**Comment by radek8** [comment]
According to the wiring diagram, it looks like the difference is in the CPU used
V3.0 - STM32G0B1RCT6
V3.0.1 - STM32F401RCT6

**Comment by tim-speight** [comment]
OK It definitely is a v3.0.  But I am wondering if it is a fake.I used the Marlin firmware distinctly for a v3.0 and the pinouts are running the wrong elements.  I wonder if there were a dodgy batch of boards BIQU manufactured in 2022.
The Box wiring with random PC0 & PC6 assignments is worrying.  How did U get on Strawberry?

**Comment by sharpstown** [comment]
I am having same issue
